src/app/settings.js

```diff
@@ -309,13 +312,14 @@ var AdvSettings = {
                     win.warn('[%s] Endpoint failed [%s]',
                         _url.hostname,
                         e.message);
-                    clearTimeout(timeout);
                     tryNextEndpoint();
                 });
+            }).setTimeout(5000, function () {
+				win.warn('[%s] Endpoint timed out',
+					_url.hostname);
+				request.abort();
+				tryNextEndpoint();
```

Comment:
I don't think it's a functional change, it's moving timeoutWrapper on line 284 and the timeout itself on line 318 and combining them. 

I'm not sure why timeoutWrapper() exists, as I don't think `request` ever changes in this context, so there's no reason for the function-in-function stuff.